### ISSUE TITLE
bare-except doesn't trigger if except raises.

### ISSUE THREAD

**Issue by natedogith1** [PyCQA/pylint#3656]
bare-except doesn't trigger if the except body directly contains a raise call.

pylint 2.4.4
astroid 2.3.3
Python 3.7.3+ (heads/3.7:e64d21b187, Apr 23 2019, 10:57:08)

```python
def foo(arg):
    try:
        pass
    except: # doesn't trigger bare-except
        raise Exception()
```
```python
def foo(arg):
    try:
        pass
    except: # doesn't trigger bare-except
        if arg:
            return
        raise Exception()
```
```python
def foo(arg):
    try:
        pass
    except: # does trigger bare-except
        if not arg:
            raise Exception()
        return
```

**Comment by AWhetter** [comment]
It looks like the code that causes this behaviour has been around for a long time. I think if there is a bare `raise` with no arguments then it makes sense to not show the warning. But for the cases you've given, a bare-except doesn't make sense over using `except Exception`.

**Comment by clavedeluna** [comment]
Reading back thru this and I think all test cases are valid to always raise bare-except

> think if there is a bare raise with no arguments then it makes sense to not show the warning.

not sure I agree with this... I actually don't understand why the [current code](https://github.com/PyCQA/pylint/blob/main/pylint/checkers/exceptions.py#L545) means that this code

```
def foo(arg):
    try:
        pass
    except:  # doesn't trigger bare-except
        raise TypeError()
```

I personally would still want to have that flagged!

**Comment by e-gebes** [comment]
> Reading back thru this and I think all test cases are valid to always raise bare-except
> 
> > think if there is a bare raise with no arguments then it makes sense to not show the warning.
> 
> not sure I agree with this... 

I think @AWhetter meant something like this:

```
def foo(arg):
    try:
        func()
    except:
        something()
        raise
```
I do not mind much about this specific case with a *bare raise*, I think we all agree that when there is a `raise SomeException` that the `bare-except` warning should appear.
